python/ray/train/batch_predictor.py

```diff
@@ -284,17 +295,27 @@ def __call__(self, input_batch: DataBatchType) -> DataBatchType:
         ray_remote_args["num_cpus"] = num_cpus_per_worker
         ray_remote_args["num_gpus"] = num_gpus_per_worker
 
-        if separate_gpu_stage and num_gpus_per_worker > 0:
-            preprocessor = self.get_preprocessor()
-            if preprocessor:
-                # Set the in-predictor preprocessing to a no-op when using a separate
-                # GPU stage. Otherwise, the preprocessing will be applied twice.
-                override_prep = BatchMapper(
-                    lambda x: x, batch_format=predict_stage_batch_format
-                )
-                # preprocessor.transform will break for DatasetPipeline due to
-                # missing _dataset_format()
+        preprocessor = self.get_preprocessor()
+
+        override_prep = None
+        if preprocessor:
+            # TODO: Delegate separate_gpu_stage flag to Datasets.
+            if not separate_gpu_stage and num_gpus_per_worker > 0:
+                override_prep = preprocessor
```

Comment:
do we actually have a test for this branch ? default separate_gpu_stage is always True.

Reply:
yeah we have one here https://github.com/ray-project/ray/blob/master/python/ray/train/tests/test_batch_predictor.py#L105